corehq/apps/email/forms.py

```diff
@@ -0,0 +1,164 @@
+import copy
+import re
+
+from crispy_forms import layout as crispy
+from crispy_forms.bootstrap import InlineField, StrictButton
+from crispy_forms.layout import Div
+from django import forms
+from django.forms import CharField, ChoiceField
+from django.forms.forms import Form
+from django.utils.safestring import mark_safe
+from django.utils.translation import gettext as _, gettext_noop
+
+from corehq.apps.email.models import SQLEmailSMTPBackend
+from corehq.apps.email.util import get_email_backend_classes
+from corehq.apps.hqwebapp import crispy as hqcrispy
+from corehq.apps.hqwebapp.crispy import HQFormHelper
+from corehq.apps.userreports.exceptions import ValidationError
+from corehq.apps.users.models import CouchUser
+
+
+class InitiateAddEmailBackendForm(Form):
+    action = CharField(
+        initial='new_backend',
+        widget=forms.HiddenInput(),
+    )
+    hq_api_id = ChoiceField(
+        required=False,
+        label="Gateway Type",
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(InitiateAddEmailBackendForm, self).__init__(*args, **kwargs)
+
+        backend_classes = InitiateAddEmailBackendForm.email_backend_classes()
+
+        backend_choices = []
+        for api_id, klass in backend_classes.items():
+            friendly_name = klass.get_generic_name()
+            backend_choices.append((api_id, friendly_name))
+
+        backend_choices = sorted(backend_choices, key=lambda backend: backend[1])
+        self.fields['hq_api_id'].choices = backend_choices
```

Comment:
nit: Just style a preference for list comprehension
```suggestion
        self.fields['hq_api_id'].choices = sorted([
            (api_id, klass.get_generic_name())
            for api_id, klass in backend_classes.items()],
            key=lambda backend: backend[1]
        )
```

Reply:
Done